src/matcha_ml/state/matcha_state.py

```diff
@@ -80,6 +79,25 @@ class MatchaState:
 
     components: List[MatchaStateComponent]
 
+    def get_component(self, resource_name: str) -> Optional[MatchaStateComponent]:
+        """Get a component of the state given a resource name.
+
+        Args:
+            resource_name (str): the components resource name
+
+        Returns:
+            MatchaStateComponent: the state component matching the resource name parameter.
```

Comment:
Update to :`Optional[MatchaStateComponent]`

Reply:
Updated, thanks.